### PR TITLE
Add `interface{}` to `any` replacement to `make fmt`, exclude `*.pb.go`

### PR DESCRIPTION
**Pull request by silverwind**
Since https://github.com/go-gitea/gitea/pull/25686, a few `interface{}` have sneaked into the codebase. Add this replacement to `make fmt` to prevent this from happening again.

Ideally a linter would do this, but I haven't found any suitable.

### CODE REVIEW

**Comment by silverwind** [comment]
`getGoVersion` is removed because gofumpt already defaults to the version in go.mod:

```
-lang       str    target Go version in the form "1.X" (default from go.mod)
```

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Don't do this for some auto generated file like `routers/api/actions/artifact.pb.go`

image

**Comment by silverwind** [comment]
How to identify generated code? What is this `.pb.go` extension?

**Comment by silverwind** [comment]
.pb.go seems to be protobuf specific for generated go code: https://protobuf.dev/reference/go/go-generated/

**Comment by silverwind** [comment]
Done, those will not be formatted or linted anymore and I marked them as generated in the gitattributes as well.

**Comment by silverwind** [comment]
BTW there is [this quip](https://golangci-lint.run/usage/configuration/) in golangci docs:

>  There is no need to include all autogenerated files,
>  we confidently recognize autogenerated files.

Imho, I would not rely on undocumented magic, so the exclude rule stays.

File: .golangci.yml [hunk 1]
```diff
@@ -93,6 +93,7 @@ issues:
   max-issues-per-linter: 0
   max-same-issues: 0
   exclude-dirs: [node_modules, public, web_src]
+  exclude-files: ["\\.pb\\.go$"]
```
Comment (silverwind) [inline .golangci.yml]:
```suggestion
  exclude-files: [".*\\.my\\.go$"]
```

File: .golangci.yml [hunk 2]
```diff
@@ -93,6 +93,7 @@ issues:
   max-issues-per-linter: 0
   max-same-issues: 0
   exclude-dirs: [node_modules, public, web_src]
+  exclude-files: [".*\\.my\\.go$"]
```
Comment (silverwind) [inline .golangci.yml]:
```suggestion
  exclude-files: [".*\\.pb\\.go$"]
```

**Review by lafriks — APPROVED** [review]

File: .golangci.yml [hunk 3]
```diff
@@ -93,6 +93,7 @@ issues:
   max-issues-per-linter: 0
   max-same-issues: 0
   exclude-dirs: [node_modules, public, web_src]
+  exclude-files: [".*\\.pb\\.go$"]
```
Comment (silverwind) [inline .golangci.yml]:
```suggestion
```

**Comment by silverwind** [comment]
> BTW there is [this quip](https://golangci-lint.run/usage/configuration/) in golangci docs:
> 
> > There is no need to include all autogenerated files,
> > we confidently recognize autogenerated files.
> 
> I would not rely on such undocumented magic, so the exclude rule stays.

Decided with going with the defaults now. `.pb.go` is excluded in golangci by all current indications, so it's better to go with the defaults, even though I disagree that golangci even has such opinionated defaults.

**Pull request merged**